test/dynamo/test_generator.py

```diff
@@ -663,6 +664,395 @@ def fn(t):
+        self.assertEqual(L, [1, -123, 2])
+
+    def test_close_capture_GeneratorExit_return(self):
+        z = 0
+
+        def whoo(t):
+            nonlocal z
+            try:
+                z += 1
+                yield t.sin()
+                yield t.cos()
+            except GeneratorExit:
+                z += 10
+                return t.tan()  # noqa: B901
+            finally:
+                z += 100
+
+        @torch.compile(backend="eager", fullgraph=True)
+        def fn(t):
+            nonlocal z
+            gen = whoo(t)
+            i = next(gen)
+            y = gen.close()
+            return (i, y)
+
+        t = torch.randn(2)
+        (i, y) = fn(t)
+        self.assertEqual(i, t.sin())
+        self.assertEqual(y, t.tan())
+        self.assertEqual(z, 111)
+
+    def test_close_capture_GeneratorExit(self):
+        z = 0
+
+        def whoo(t):
+            nonlocal z
+            try:
+                yield t.sin()
+                yield t.cos()
+            except GeneratorExit:
+                yield t.tan()
+            finally:
+                z = 1
+
+        @torch.compile(backend="eager", fullgraph=True)
+        def fn(t):
+            nonlocal z
+            gen = whoo(t)
+            i = next(gen)
+            gen.close()
+            return i
+
+        t = torch.randn(2)
+        # This should actually be RuntimeError("generator ignored GeneratorExit")
+        # but Dynamo swallow the exception and raises Unsupported instead
+        with self.assertRaisesRegex(Unsupported, "Observed exception"):
+            fn(t)
+
+    def test_close_capture_and_reraise_GeneratorExit(self):
+        L = []
+        z = 0
+
+        def whoo(t):
+            nonlocal z
+            try:
+                L.append(1)
+                yield t.sin()
+                yield t.cos()
+            except GeneratorExit:
+                L.append(z)
+                z = -1
+                raise
+            finally:
+                L.append(z)
+
+        @torch.compile(backend="eager", fullgraph=True)
+        def fn(t):
+            nonlocal z
+            gen = whoo(t)
+            i = next(gen)
+            z = -123
+            gen.close()
+            L.append(456)
+            return i
+
+        t = torch.randn(2)
+        y = fn(t)
+        self.assertEqual(y, t.sin())
+        self.assertEqual(L, [1, -123, -1, 456])
+
+    def test_close_capture_and_reraise_RuntimeError(self):
+        def whoo(t):
+            try:
+                yield t.sin()
+                yield t.cos()
+            except GeneratorExit as e:
+                raise RuntimeError from e
+            finally:
+                pass
+
+        @torch.compile(backend="eager", fullgraph=True)
+        def fn(t):
+            gen = whoo(t)
+            i = next(gen)
+            gen.close()
+            return i
+
+        t = torch.randn(2)
+        with self.assertRaises(RuntimeError):
+            fn(t)
+
+    @unittest.expectedFailure
```

Comment:
Did you want this to pass or is this intentionally expectedFailure?

Reply:
It's intentional. It needs `CLEANUP_THROW` which is implemented in a follow-up PR.